### PR TITLE
resolves #87 Improve web accessibility for screen reader

### LINKED ISSUE #87: Improve web accessibility for screen reader
The purpose of this issue is to improve the tock-react-kit accessibility for screen reader. Following an audit on our web chatbot, we had several issues on the code generated by the tock-react-kit. Here are the issues detected : 

1. In the chat container, some attributes are missing
    - aria-live='polite'
    - aria-atomic='false'
    - aria-relevant='additions'
2. Messages are generated with div elements, it is better to generate it with list elements (`<ul><li>`) with li elements for each message.
3. Header elements for titles and subtitles in card can be tricky to read for screen reader, use another element to show it.
4. Use list elements for quick replies instead of div elements
5. Image in cards should have an attribute alternative which describes the content of the image or at least the purpose of the card. It is hard to describe it with just a title.
6. In the Carousel container, some attributes are missing
    - role='group'
    - aria-roldescription='Carousel' (Maybe make it dynamically to support other languages)
7. For each slide of a Carousel, somes attributes are missing
    - role='group'
    - aria-roledescription='Slide' or 'Result' (Maybe make it dynamically to support other languages)
8. On previous/next image buttons of Carousel, some attributes are missing
    - role='img'
    - aria-label='Previous results' or 'Next results' (Maybe make it dynamically to support other languages)
    - focusable='false'
9. Add on hidden slides in Carousel the attributes dynamically : 
    - aria-hidden='true'
    - tabIndex='-1'
10. Add attributes to the sendMessage img button
    - role='img'
    - aria-label='Send message' (Maybe make it dynamically to support other languages)
    - focusable='false'

### PR DESCRIPTION
**Pull request by correi-f**
resolves #87 Improve web accessibility for screen reader

### CODE REVIEW

File: src/components/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -127,17 +132,17 @@ const Carousel: (props: { children?: ReactElement[] }) => JSX.Element = ({
...
+      <ItemContainer ref={ref.container} role='group' aria-roledescription={accessibility?.carouselRoleDescription || 'Carousel'}>
         {children?.map((child, i) =>
-          React.cloneElement(child, { ref: ref.items[i] }, undefined),
+          React.cloneElement(child, { ref: ref.items[i], roleDescription: accessibility?.slideRoleDescription }, undefined),
         )}
       </ItemContainer>
       {rightVisible && (
         <Next onClick={next}>
-          <ArrowRightCircle size={`calc(${theme.typography.fontSize} * 2)`} />
+          <ArrowRightCircle size={`calc(${theme.typography.fontSize} * 2)`} role='img' aria-label={accessibility?.previousCarouselButtonLabel || 'Next slides'} focusable='false'/>
```
Comment (francoisno) [inline src/components/Carousel/Carousel.tsx:145]: Hi @correi-f it should be `nextCarouselButtonLabel` there - probably a copy-paste mistake ;)

**Review by francoisno — CHANGES_REQUESTED** [review]
A small mistake to fix, but anyway great work and addition thx @correi-f !

File: src/components/Carousel/Carousel.tsx (hunk 1 above)
Comment (correi-f) [reply src/components/Carousel/Carousel.tsx]: Done

File: README.md [hunk 2]
```diff
@@ -255,6 +255,38 @@ A `TockTheme` can be used as a value of a `ThemeProvider` of [`emotion-theming`]
...
+Example :
+
+```js
+renderChat(
+    document.getElementById('chat'),
+    '<TOCK_BOT_API_URL>',
+    undefined,
+    {},
+    { accessibility: {
+        carouselRoleDescription: 'Carrousel',
```
Comment (Fabilin) [inline README.md:284]:
```suggestion
        carouselRoleDescription: 'Carousel',
```

**Comment by francoisno** [comment]
Merged manually.

**Pull request closed without merging**